core/test-set.ts

```diff
@@ -1,44 +1,51 @@
-import { GlobHelper, TestLoader } from "./_core";
+import { GlobHelper, TestLoader, FileRequirer } from "./_core";
 import { ITestFixture } from "./_interfaces";
 
 const path = require("path");
 
 export class TestSet {
 
-  private _testFixtures: Array<ITestFixture> = [];
+   private _testFixtures: Array<ITestFixture> = [];
 
-  public get testFixtures(): Array<ITestFixture> {
-    return this._testFixtures;
-  }
+   public get testFixtures(): Array<ITestFixture> {
+      return this._testFixtures;
+   }
 
-  public constructor(private _testLoader: TestLoader, private _globHelper: GlobHelper) { }
+   public constructor(private _testLoader: TestLoader, private _globHelper: GlobHelper) { }
 
-  public addTestsFromFiles (testFileLocation: string): void
-  public addTestsFromFiles (testFileLocations: Array<string>): void
-  public addTestsFromFiles (testsFileLocations: string | Array<string>) {
+   public static create(): TestSet {
+      const fileRequirer = new FileRequirer();
+      const testLoader = new TestLoader(fileRequirer);
+      const globHelper = new GlobHelper();
+      return new TestSet(testLoader, globHelper);
+   }
 
-    if (typeof testsFileLocations === "string") {
-      testsFileLocations = [ <string>testsFileLocations ];
-    }
+   public addTestsFromFiles (testFileLocation: string): void
+   public addTestsFromFiles (testFileLocations: Array<string>): void
+   public addTestsFromFiles (testsFileLocations: string | Array<string>) {
```

Comment:
I believe this is to handle older versions of TypeScript that don't support this very well, in that they need to have the intended uses and also a union of all of them. This seems to work nicely for 2.0 but not so much for lower so wanted to be nice to the peeps who haven't got to upgrade yet :)